pkg/clusteroperator/clusteroperator_test.go

```diff
@@ -148,12 +149,123 @@ func TestClusterOperatorGet(t *testing.T) {
 	}
 }
 
+func TestHasDesiredVersion(t *testing.T) {
+	testCases := []struct {
+		desiredVersion      string
+		expectedOutput      bool
+		testClusterOperator *Builder
+		expectedError       error
+	}{
+		{
+			desiredVersion:      "",
+			expectedOutput:      false,
+			testClusterOperator: buildValidClusterOperatorBuilder(buildClusterOperatorClientWithDummyObject()),
+			expectedError:       fmt.Errorf("desiredVersion can't be empty"),
+		},
+		{
+			desiredVersion: "4.14.0",
+			expectedOutput: true,
+			testClusterOperator: buildFakeClusterOperatorWithVersions(buildClusterOperatorClientWithDummyObject(),
+				[]string{"4.13.2", "4.14.0"}),
+			expectedError: nil,
+		},
+		{
+			desiredVersion:      "4.14.0",
+			expectedOutput:      false,
+			testClusterOperator: nil,
+			expectedError:       fmt.Errorf("error: received nil ClusterOperator builder"),
+		},
+		{
+			desiredVersion: "4.14.0",
+			expectedOutput: false,
+			testClusterOperator: buildFakeClusterOperatorWithVersions(buildClusterOperatorClientWithDummyObject(),
+				[]string{"4.13.2", "4.13.0"}),
+			expectedError: nil,
+		},
+		{
+			desiredVersion: "4.14.0",
+			expectedOutput: false,
+			testClusterOperator: buildFakeClusterOperatorWithVersions(buildClusterOperatorClientWithDummyObject(),
+				[]string{"4.13.2", "invalid"}),
+			expectedError: nil,
+		},
+		{
+			desiredVersion:      "4.14.0",
+			expectedOutput:      false,
+			testClusterOperator: buildFakeClusterOperatorWithVersions(buildClusterOperatorClientWithDummyObject(), []string{}),
+			expectedError:       fmt.Errorf("undefined cluster operator status versions"),
+		},
+		{
+			desiredVersion:      "4.14.0",
+			expectedOutput:      false,
+			testClusterOperator: buildNilClientClusterOperatorBuilder(),
+			expectedError:       fmt.Errorf("ClusterOperator builder cannot have nil apiClient"),
+		},
+	}
+	for _, testCase := range testCases {
+		result, err := testCase.testClusterOperator.HasDesiredVersion(testCase.desiredVersion)
+		assert.Equal(t, testCase.expectedOutput, result)
+		assert.Equal(t, testCase.expectedError, err)
+	}
+}
+
+func TestVerifyClusterOperatorsVersion(t *testing.T) {
```

Comment:
Please add new file **list_test.go** and move this function it the new file.

Reply:
Added, thanks